src/Framework/LogInterpolatedStringHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Build.Framework
+{
+    [InterpolatedStringHandler]
+    public ref struct LogInterpolatedStringHandler
+    {
+        private readonly char[] buffer;
+        private int position = 0;
+        private int argPosition = 0;
+
+        public object?[] Arguments { get; } = Array.Empty<object?>();
+
+        public LogInterpolatedStringHandler(int literalLength, int formattedCount)
+        {
+            if (formattedCount > 99)
```

Comment:
I don't think we want to ship with this limitation. Should be easy to support any number. I was going to write the function to calculate `bufferSize` but ChatGPT is not available at the moment :wink:

Reply:
I like the idea. It's easy to compute number of digits in ```formattedCount``` with ```Log10```, but in that case, I would drop the exact buffer size computation for sake of simplicity and runtime complexity. I believe that cases with 10 and more arguments are very rare and saving few bytes doesn't worth it.

Reply:
By `Log10` I assume you mean `while (formattedCount > 0) { numOfDigits++; formattedCount /= 10 }`, not the actual floating point `Math.Log10`. Floating point arithmetic can be very costly on some CPUs.

Reply:
I used ```Log10``` for cases where we have more than 10 arguments. I still believe that these cases are very rare. I don't have an opportunity to do measurements on various CPUs, but I found perf results for this task: https://i.stack.imgur.com/YUfLu.png. It was published few years ago. If you insist to use a loop, I will rewrite it :)

Reply:
Thank you! I would favor the loop but I may be too floating point averse. @rainersigwald, what do you think?

Notes:
- The benchmark probably used random integer values while ours are expected to be very small.
- With the loop there would be no need for the `if (formattedCount < 10)` special case.

Reply:
Okay, no problem. I removed ```Log10``` :)